src/main/java/uk/gov/ons/census/fwmt/outcomeservice/converter/spg/SpgLinkedQidProcessor.java

```diff
@@ -0,0 +1,80 @@
+package uk.gov.ons.census.fwmt.outcomeservice.converter.spg;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+import uk.gov.ons.census.fwmt.outcomeservice.config.GatewayOutcomeQueueConfig;
+import uk.gov.ons.census.fwmt.outcomeservice.converter.SpgOutcomeServiceProcessor;
+import uk.gov.ons.census.fwmt.outcomeservice.data.GatewayCache;
+import uk.gov.ons.census.fwmt.outcomeservice.data.GatewayCache.GatewayCacheBuilder;
+import uk.gov.ons.census.fwmt.outcomeservice.dto.FulfilmentRequestDto;
+import uk.gov.ons.census.fwmt.outcomeservice.dto.SpgOutcomeSuperSetDto;
+import uk.gov.ons.census.fwmt.outcomeservice.message.GatewayOutcomeProducer;
+import uk.gov.ons.census.fwmt.outcomeservice.service.impl.GatewayCacheService;
+import uk.gov.ons.census.fwmt.outcomeservice.template.TemplateCreator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.CESPG_ADDRESS_NOT_VALID_OUTCOME_SENT;
+import static uk.gov.ons.census.fwmt.outcomeservice.config.GatewayEventsConfig.CESPG_OUTCOME_SENT;
+import static uk.gov.ons.census.fwmt.outcomeservice.enums.EventType.LINKED_QID;
+import static uk.gov.ons.census.fwmt.outcomeservice.enums.SurveyType.spg;
+
+@Component("LINKED_QID")
+public class SpgLinkedQidProcessor implements SpgOutcomeServiceProcessor {
+
+  @Autowired
+  private GatewayOutcomeProducer gatewayOutcomeProducer;
+
+  @Autowired
+  private GatewayEventManager gatewayEventManager;
+
+  @Autowired
+  private GatewayCacheService gatewayCacheService;
+
+  private boolean isQuestionnaireLinked(FulfilmentRequestDto fulfilmentRequest) {
+    return (fulfilmentRequest.getQuestionnaireType() == null);
+  }
+
+  @Override
+  public UUID process(SpgOutcomeSuperSetDto outcome, UUID caseIdHolder) throws GatewayException {
+    if (outcome.getFulfillmentRequests() == null) {
+      return caseIdHolder;
+    }
+    for (FulfilmentRequestDto fulfilmentRequest : outcome.getFulfillmentRequests()) {
+      if (isQuestionnaireLinked(fulfilmentRequest)) {
+        String eventDateTime = outcome.getEventDate().toString();
```

Comment:
Use SimpleDateFormatter to convert to correct format YYYY-MM-DD THH:MM:SS.ssssssZ

Reply:
Done.